src/lib/interactionManager/interaction.ts

```diff
@@ -138,10 +138,18 @@ export class Interaction {
 
   public async processInteractionToken(token: JSONWebToken<JWTEncodable>) {
     if (!this.participants) {
+      // TODO what happens if the signer isnt resolvable
+      const requester = await this.ctx.registry.resolve(token.signer.did)
       this.participants = {
-        us: this.ctx.identityWallet.identity,
-        them: await this.ctx.registry.resolve(token.signer.did),
```

Comment:
this logic always assumes that this SDK instance is the client/responder/user. for services obviously this is not the case, so it should assign based on order of reciept